### PR TITLE
Introduce new PCI framework

### PR DESCRIPTION
**Pull request by lipengfei28**

### CODE REVIEW

File: drivers/pci/pci_drivers.c [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: pci_register_drivers
+ *
+ * Description:
+ *   Register all the pci drivers to pci bus
+ ****************************************************************************/
```
Comment (raiden00pl) [inline drivers/pci/pci_drivers.c:40]:
```suggestion
 *   Register all the pci drivers to pci bus
 *
 ****************************************************************************/
```

**Review by raiden00pl — CHANGES_REQUESTED** [review]
Please fix all function headers according to this issue:
https://github.com/apache/nuttx/issues/12133
The end of the header should be preceded by a new line like this:
```
 *
 ****************************************************************************/
```

File: boards/x86_64/intel64/qemu-intel64/configs/earlyfb/defconfig [hunk 2]
```diff
@@ -58,6 +54,3 @@ CONFIG_SYSTEM_CLE=y
 CONFIG_SYSTEM_NSH=y
 CONFIG_SYSTEM_TIME64=y
 CONFIG_USEC_PER_TICK=1
-CONFIG_VIRT=y
-CONFIG_VIRT_QEMU_EDU=y
-CONFIG_VIRT_QEMU_PCI_TEST=y
```
Comment (raiden00pl) [inline boards/x86_64/intel64/qemu-intel64/configs/earlyfb/defconfig:63]:
```
CONFIG_PCI_QEMU_TEST=y
CONFIG_PCI_QEMU_EDU=y
```

File: boards/x86_64/intel64/qemu-intel64/configs/pcitest/defconfig [hunk 3]
```diff
@@ -63,6 +59,3 @@ CONFIG_SYSTEM_CLE=y
 CONFIG_SYSTEM_NSH=y
 CONFIG_SYSTEM_TIME64=y
 CONFIG_USEC_PER_TICK=1
-CONFIG_VIRT=y
```
Comment (raiden00pl) [inline boards/x86_64/intel64/qemu-intel64/configs/pcitest/defconfig:66]:
```
CONFIG_PCI_QEMU_TEST=y
CONFIG_PCI_QEMU_EDU=y
```

File: boards/x86_64/intel64/qemu-intel64/configs/pcitest/defconfig [hunk 4]
```diff
@@ -27,10 +27,6 @@ CONFIG_BOOT_RUNFROMEXTSRAM=y
 CONFIG_BUILTIN=y
 CONFIG_CONSOLE_SYSLOG=y
 CONFIG_DEBUG_FEATURES=y
-CONFIG_DEBUG_PCI=y
```
Comment (raiden00pl) [inline boards/x86_64/intel64/qemu-intel64/configs/pcitest/defconfig:30]: It's better to leave debug options enabled for `pcitest` configuration

Comment (lipengfei28) [reply boards/x86_64/intel64/qemu-intel64/configs/pcitest/defconfig:30]: done

File: boards/x86_64/intel64/qemu-intel64/configs/pcitest/defconfig (hunk 3 above)
Comment (lipengfei28) [reply boards/x86_64/intel64/qemu-intel64/configs/pcitest/defconfig:66]: done

File: boards/x86_64/intel64/qemu-intel64/configs/earlyfb/defconfig (hunk 2 above)
Comment (lipengfei28) [reply boards/x86_64/intel64/qemu-intel64/configs/earlyfb/defconfig:63]: done

**Review by raiden00pl — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**